src/app.js

```diff
@@ -1,21 +1,46 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const http = require('http');
+const url = require('url');
+
+const PORT = process.env.PORT || 3000;
+
+const server = http.createServer((req, res) => {
+  res.setHeader('Content-Type', 'application/json');
+
+  const normalizedUrl = new url.URL(req.url, `http://${req.headers.host}`);
+
+  const result = {
+    'parts': [],
+    'query': {
+    },
+  };
+
+  const pathname = normalizedUrl.pathname;
+  const parts = pathname.split('/').filter((el) => el !== '');
+
+  result.parts.push(...parts);
+
+  if (normalizedUrl.searchParams) {
+    for (const [key, value] of normalizedUrl.searchParams.entries()) {
+      if (result.query.hasOwnProperty(key)) {
+        if (Array.isArray(result.query[key])) {
+          result.query[key].push(value);
+        } else {
+          result.query[key] = [result.query[key], value];
+        }
+      } else {
+        result.query[key] = value;
+      }
+    }
+  }
```

Comment:
it should be working just fine, if you just set Object.fromEntries(normalizedUrl.searchParams.entries()) as value of query inside of result object
I'd also consider using default object, which you will return as a JSON using JSON,stringify()